openedxscorm/scormxblock.py

```diff
@@ -331,15 +362,28 @@ def extract_package(self, package_file):
     def index_page_url(self):
         if not self.package_meta or not self.index_page_path:
             return ""
-        folder = self.extract_folder_path
-        if self.storage.exists(
-            os.path.join(self.extract_folder_base_path, self.index_page_path)
-        ):
-            # For backward-compatibility, we must handle the case when the xblock data
-            # is stored in the base folder.
-            folder = self.extract_folder_base_path
-            logger.warning("Serving SCORM content from old-style path: %s", folder)
-        return self.storage.url(os.path.join(folder, self.index_page_path))
+
+        querystring_auth_setting = self.xblock_settings.get('SCORM_S3_QUERY_AUTH')
+        if querystring_auth_setting is False:
+            folder = self.extract_folder_path
+            if self.storage.exists(
+                os.path.join(self.extract_folder_base_path, self.index_page_path)
+            ):
+                # For backward-compatibility, we must handle the case when the xblock data
+                # is stored in the base folder.
+                folder = self.extract_folder_base_path
+                logger.warning("Serving SCORM content from old-style path: %s", folder)
+            return self.storage.url(os.path.join(folder, self.index_page_path))
+        else:
```

Comment:
The XBlock should not have to be aware of all this logic. Instead, this function should not be modified and all the logic should be moved to the `self.storage.url` function. Since `self.storage` will be an instance of `S3ScormStorage` it will be possible to override the `url` method there.